src/test/java/core/basesyntax/service/FruitTransactionParserTest.java

```diff
@@ -0,0 +1,71 @@
+package core.basesyntax.service;
+
+import core.basesyntax.models.FruitTransaction;
+import core.basesyntax.service.impl.FruitTransactionParserImpl;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FruitTransactionParserTest {
+    private static FruitTransactionParser parser;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        parser = new FruitTransactionParserImpl();
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void getFruitTransactionsList_nullInput_NotOk() {
+        parser.getFruitTransactionsList(null);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void getFruitTransactionsList_DataContainsNull_notOk() {
+        List<String> listWithNull = new ArrayList<>();
+        listWithNull.add(null);
+        parser.getFruitTransactionsList(listWithNull);
+    }
+
+    @Test
+    public void getFruitTransactionsList_Ok() {
+        FruitTransaction bananaFruitTransaction = new FruitTransaction();
+        bananaFruitTransaction.setOperation(FruitTransaction.Operation.BALANCE);
+        bananaFruitTransaction.setFruit("banana");
+        bananaFruitTransaction.setQuantity(10);
+        FruitTransaction appleFruitTransaction = new FruitTransaction();
+        appleFruitTransaction.setOperation(FruitTransaction.Operation.RETURN);
+        appleFruitTransaction.setFruit("apple");
+        appleFruitTransaction.setQuantity(5);
+        List<String> list = new ArrayList<>();
+        list.add("fruit,quantity");
+        list.add("b,banana,10");
+        list.add("r,apple,5");
+        List<FruitTransaction> expected = List.of(
+                bananaFruitTransaction, appleFruitTransaction);
+        List<FruitTransaction> actual = parser.getFruitTransactionsList(list);
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void getFruitTransactionsList_InvalidData_NotOk() {

```

Comment:
```suggestion
    public void getFruitTransactionsList_InvalidData_ok() {
```
not ok are only cases where we expect an exception